### PR TITLE
fix: handle deep project paths in sidebar width and slug parsing

### LINKED ISSUE #327: fix(ui): sidebar maxWidth is too constrained for deeply nested project paths
## Problem

The left selector sidebar feels locked when working with deeply nested project paths. With an organizational pattern like `~/Projects/{GitHost}/{GitOrg}/{RepoName}` (5+ path segments), project names overflow the sidebar invisibly even when the user drags it to the maximum allowed width.

The sidebar IS technically resizable — `src/components/ProjectTree/index.tsx:972` renders a 12px transparent drag handle — but the `maxWidth: 480` ceiling in `src/App.tsx:232` makes the panel hit its limit before long names fit. Users with this kind of path organization often don't realize the panel was ever resizable.

## Diagnosis

`src/App.tsx:229-234` configures the resize hook:

```ts
const {
  width: sidebarWidth,
  ...
} = useResizablePanel({
  defaultWidth: 256,
  minWidth: 200,
  maxWidth: 480,
  storageKey: "sidebar-width",
});
```

`src/components/ProjectTree/components/ProjectItem.tsx:125-143` renders the project name with `whitespace-nowrap` and `flex-1` but no `truncate` or `overflow-hidden`, so once the name exceeds the available width it just spills behind the panel edge with no ellipsis indicator.

At 480px max, after subtracting tree indentation, expand chevron, folder icon, provider badge, and session count, there is roughly 280-320px of horizontal space for the project name. Paths like `internal.github.acme.com/dumb-department-name/dummy-app-microservice-apple` need substantially more.

## Suggested Fix

Raise `maxWidth` from `480` to `800` in `src/App.tsx:232`. Default (256) and minimum (200) remain unchanged.

Existing persisted widths in localStorage are still valid: the hook's init guard at `src/hooks/useResizablePanel.ts:86` checks `parsed >= minWidth && parsed <= maxWidth`, and any previously persisted value ≤480 is still ≥200, so it passes through unchanged. Users only see the new ceiling when they next drag the handle.

Two related improvements that are intentionally out of scope here (could be follow-up issues if they prove necessary):

- Making the 12px resize handle visible at rest (currently transparent until hover) so it is discoverable.
- Adding `truncate` + ellipsis to long project names so overflow is visible even before the user resizes.

This fix is just the ceiling raise. Confirmed approach.

## Labels

bug, enhancement, ui

### LINKED ISSUE #328: fix(slug): extract_project_name returns encoded suffix for paths deeper than 3 segments
## Problem

Project names in the sidebar are mangled when the filesystem path has more than three segments. For example, a project at `~/Projects/GitHub/my-org/my-repo` shows up as `Projects-GitHub-my-org-my-repo` in the project tree instead of `my-repo`. With a deeper, more realistic path like `~/Projects/internal.github.acme.com/dumb-department-name/dummy-app-microservice-apple`, the displayed name becomes `Projects-internal.github.acme.com-dumb-department-name-dummy-app-microservice-apple` — essentially the encoded slug with the leading two segments stripped.

This happens for every Claude project a user owns whose path is deeper than three directory levels.

## Diagnosis

`src-tauri/src/utils.rs:54-65` (`extract_project_name`):

```rust
pub fn extract_project_name(raw_project_name: &str) -> String {
    if raw_project_name.starts_with('-') {
        let parts: Vec<&str> = raw_project_name.splitn(4, '-').collect();
        if parts.len() == 4 {
            parts[3].to_string()
        } else {
            raw_project_name.to_string()
        }
    } else {
        raw_project_name.to_string()
    }
}
```

`splitn(4, '-')` produces at most four pieces. For the slug `-Users-cameron-Projects-GitHub-org-repo`, the pieces are `["", "Users", "cameron", "Projects-GitHub-org-repo"]`. The function returns `parts[3]` verbatim — anything past the third hyphen, including hyphens that should have been path separators.

The codebase already has a correct decoder for arbitrarily deep paths: `decode_with_filesystem_check` (same file, line 225) recurses through every possible hyphen position and confirms each candidate via `symlink_metadata`, so it handles both deep nesting AND directory names that contain hyphens (e.g. `dumb-department-name`). It's just not called from `extract_project_name`.

## Suggested Fix

Delegate to `decode_with_filesystem_check` first, fall back to the existing `splitn(4)` heuristic when the path no longer exists on disk:

```rust
pub fn extract_project_name(raw_project_name: &str) -> String {
    if let Some(stripped) = raw_project_name.strip_prefix('-') {
        // Filesystem-checked decoding handles arbitrarily deep paths AND
        // project names that themselves contain hyphens.
        if let Some(decoded) = decode_with_filesystem_check(stripped) {
            if let Some(leaf) = Path::new(&decoded).file_name() {
                return leaf.to_string_lossy().to_string();
            }
        }
        // Fallback for paths that no longer exist on disk (deleted projects).
        let parts: Vec<&str> = raw_project_name.splitn(4, '-').collect();
        if parts.len() == 4 {
            return parts[3].to_string();
        }
    }
    raw_project_name.to_string()
}
```

This is the same logic flow as in `decode_project_path` (line 181), which already uses filesystem decoding correctly for the `actual_path` field. The fix makes `name` consistent with `actual_path`.

The existing six unit tests for `extract_project_name` continue to pass — their inputs are paths that don't exist on disk, so they fall through to the heuristic branch and return the same answer as before. Two new tests cover the deep-path cases, including one with hyphens in segment names to validate the disambiguation. Confirmed approach.

## Labels

bug, rust

### PR DESCRIPTION
**Pull request by cameronsjo**
## Summary

Two small, related fixes for users with deeply nested project paths.

- **`src/App.tsx`**: raise `useResizablePanel` `maxWidth` from `480` to `800` so the left sidebar can stretch wide enough to show long project names. Default (256) and min (200) unchanged; existing persisted widths in localStorage stay valid.
- **`src-tauri/src/utils.rs`**: `extract_project_name` now delegates to the existing `decode_with_filesystem_check` helper before falling back to the original `splitn(4, '-')` heuristic. The heuristic returns `parts[3]` verbatim for slugs with more than 3 hyphens, so a project at `~/Projects/GitHub/org/repo` would render as `"Projects-GitHub-org-repo"` in the sidebar. The filesystem-checked decoder disambiguates correctly, including project names that themselves contain hyphens (e.g. `dumb-department-name`).

Both fixes preserve existing behavior for shallow paths and for projects whose directories no longer exist on disk (the heuristic remains as fallback). All existing `extract_project_name` tests continue to pass; three new tests cover the deep-path success cases and the heuristic-fallback case explicitly.

Closes #327
Closes #328

## Test plan

- [x] `cargo test -- --test-threads=1` — full suite passes (incl. 3 new tests: `test_extract_project_name_deep_path`, `test_extract_project_name_with_hyphens_in_segments` (validates the realistic case `~/Projects/internal.github.acme.com/dumb-department-name/dummy-app-microservice-apple` → `dummy-app-microservice-apple`), and `test_extract_project_name_fallback_when_path_missing` (makes the heuristic-fallback branch coverage explicit instead of relying on dev-filesystem state))
- [x] `cargo clippy --all-targets --all-features -- -D warnings` — clean
- [x] `cargo fmt --all -- --check` — clean
- [x] `pnpm tsc --build .` — clean
- [x] `pnpm vitest run` — 785 tests pass
- [x] `pnpm lint` — clean (1 pre-existing warning unrelated to this change)
- [x] `pnpm run i18n:validate` — all keys synced across en/ko/ja/zh-CN/zh-TW
- [x] Manual: dragged sidebar to ~800px in `just dev`, confirmed long project names render correctly

## Notes

This PR intentionally does NOT include two related improvements that surfaced during investigation, in order to keep the diff focused:

- The resize handle in `ProjectTree/index.tsx:972` is `w-3` and transparent until hover, making it hard to discover.
- `ProjectItem.tsx` renders the project name with `whitespace-nowrap` and no `truncate` — long names overflow invisibly even after the ceiling raise if a user keeps the sidebar narrow.

Happy to file follow-ups if you'd like to address either.

## Summary by CodeRabbit

* **Bug Fixes**
  * More reliable project-name extraction for dash-prefixed encoded paths, returning only the leaf name and falling back to the prior heuristic when needed.

* **Style**
  * Increased sidebar maximum resize width from 480 to 800 pixels for greater layout flexibility.

* **Tests**
  * Added unit tests (including helper) covering fallback, deep-path, and hyphen-containing segment cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/jhlee0409/claude-code-history-viewer/pull/329)

### CODE REVIEW

File: src-tauri/src/utils.rs [hunk 1]
```diff
@@ -600,6 +605,66 @@ mod tests {
...
+            .expect("canonicalize temp dir")
+            .join(root_name);
+        let mut deep = root.clone();
+        for s in segments {
+            deep = deep.join(s);
+        }
+        std::fs::create_dir_all(&deep).expect("create deep tmp dir");
+        let encoded = deep.to_string_lossy().replace('/', "-");
+        (encoded, root)
+    }
```
[the remark this answers is not in the document]
Comment (cameronsjo) [reply src-tauri/src/utils.rs:638]: Good catch — fixed in f1bf7cb.

The test helper now normalizes both `/` and `\` via `replace(['/', '\\'], "-")` and inserts a leading `-` when the path doesn't start with one (Windows paths begin with a drive letter, not `/`). On macOS the change is a no-op (paths already start with `/` and contain no `\`); on Windows the encoded slug now matches Claude's leading-dash convention so `extract_project_name`'s `strip_prefix('-')` branch is exercised as intended.

@coderabbitai resolve

**Comment by jhlee0409** [comment]
Thanks — delegating to `decode_with_filesystem_check` with the `splitn(4)` fallback is the right shape, and the new tests cover the edge cases nicely. Merging now.
